packages/runtime/src/runtimes/react/components/hooks/use-builder-handshake.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useMemo } from 'react'
+import { useIsPagesRouter } from '../../../../next/hooks/use-is-pages-router'
+
+const PAGES_ROUTER_HANDSHAKE_MESSAGE = 'makeswift_preview_mode'
+const APP_ROUTER_HANDSHAKE_MESSAGE = 'makeswift_draft_mode'
+
+export function useBuilderHandshake(appOrigin: string) {
+  const isPagesRouter = useIsPagesRouter()
+
+  const handshakeInitiationMessage = useMemo(() => {
+    if (isPagesRouter) return PAGES_ROUTER_HANDSHAKE_MESSAGE
+    return APP_ROUTER_HANDSHAKE_MESSAGE
+  }, [isPagesRouter])
+
+  useEffect(() => {
+    if (window.parent !== window) {
+      window.parent.postMessage({ type: handshakeInitiationMessage }, appOrigin)
```

Comment:
nit: if we use the object format for the second parameter, it might help future readers as to what the parameter is. i.e.:

```ts
postMessage({ type }, { targetOrigin: appOrigin })
```